packages/ui/src/types/authenticator/stateMachine/authMachine.ts

```diff
@@ -25,4 +25,7 @@ export type AuthInterpreter = Interpreter<
  */
 export type AuthMachineSend = AuthInterpreter['send'];
 
-export type HubHandler = (data: HubCapsule, service: AuthInterpreter) => void;
+export type HubHandler = (
```

Comment:
Can we get this type directly from Amplify JS?

Reply:
That makes sense, my concern is that we may be exposing incorrect typing for the `HubHandler`

Reply:
In what way do you think we might be exposing the incorrect typing for HubHandler?

Reply:
That's a fair point. To be fair, the callback we provide to `Hub.listen` is a `(capsule: HubCapsule) => void`: https://github.com/aws-amplify/amplify-ui/blob/28ddfb0deb6d214d5f91427beb42fcae35b9919a/packages/ui/src/helpers/authenticator/utils.ts#L73-L75. 

This is a synchronous function that calls `defaultAuthHubHandler` (in fire-and-forget manner) under the hood.